### DISCUSSION TITLE
DayPickerInput prefers typed value over controlled value

### DISCUSSION THREAD

**Discussion by MitchRivet** [gpbl/react-day-picker#1119] [Support]
reproduction here: https://codesandbox.io/s/0yjvkzm3vn

I know this is a very specific example, but it has come up in my particular use case. In the example above, I have a DayPickerInput with a controlled value. When I type an invalid date into the input, and the value is updated (from outside of the component) the typed valid is not cleared. This is the opposite behavior when the user changes the date from the input's overlay. 

This happens because of the order of variables in the OR condition here: https://github.com/gpbl/react-day-picker/blob/master/src/DayPickerInput.js#L569

It seems like switching the order of those two variables would be the desired approach (this.state.value is the source of truth for the input's value) 

![sharedstateexample](https://user-images.githubusercontent.com/13023846/46565101-0dc41e00-c8c1-11e8-9a4f-424b52755849.gif)

**Comment by walterdn** [comment]
I believe this is the same issue as:
https://github.com/gpbl/react-day-picker/issues/804

**Comment by ivan-nejezchleb** [comment]
We have the same issue in our application.

In version 7.0.7 it works fine. 
But in 7.2 there is no way how to reset the text. It is caused by merge of https://github.com/gpbl/react-day-picker/pull/778

I propose different solution, but I am not sure, if this doesn't break some other use case
- clear inner state 'typedValue' to undefined in componentDidUpdate, when props.value has changed (or at least when has changed to some valid date).
https://github.com/gpbl/react-day-picker/blob/master/src/DayPickerInput.js#L189
